modules/ROOT/pages/6.1-release-notes.adoc

```diff
@@ -0,0 +1,1121 @@
+
+
+=== UI components, such as dialogs, would in some cases cause the esc keyup event to incorrectly trigger inside the editor
+
+In {productname} the *esc* (Escape) key can be used to exit or ‘escape’ from an active user-interface component. If a menu is open, pressing the esc key closes the menu. If a dialog is present, pressing the esc closes the dialog.
+
+{productname} responds to the `KeyDown` event and exits the active element when the *esc* key is pressed. Previously, however, the consequent `KeyUp` event (generated when the user stops pressing the *esc* key) was propogated to the {productname} editor. This should not have happened.
+
+The `KeyDown` event does not propagate outside the component listening for it and neither should the `KeyUp` event.
+
+As of this update, this is the behavior: pressing the *esc* key closes or exits an active user-interface component and neither the `KeyDown` or `KeyUp event are propogated outside the component listening for these events.
+
+
+=== Selection direction was not stored/restored when getting/setting selection bookmarks
+
+_Selection direction_ is a real-time state that ‘knows’ where the current text selection began and, consequently, ‘knows’ which direction of pointer movement or which arrow key selects more text or de-selects already selected text.
+
+For example, consider a document containing words written in a left-to-right script and an insertion point set at the end of a word. A user then holds down the Shift key and presses the left-arrow key. The selection direction for this moment and this action is set by said action. While the Shift key is held down, pressing the left-arrow key adds letters to the selection and pressing the right-arrow key removes letters from the selection.
+
+When a {productname} document had *View > Show invisible characters* checked and text was selected in this manner, and the user paused for a second or two, pausing the selection extentsion, the `moveToBookmark` function copied the current selection into a bookmark and then restored the selection in the document.
+
+In doing this, however, the selection direction was lost, because `movetoBookmark` did not store the selection direction state.
+
+When a user re-started the selection process by pressing the right arrow key, {productname} extended the selection from the right-most already-selected letter rather than, as expected, removing the left-most already-selected letter from the selection.
+
+As of this update, `moveToBookmark` now stores and restores selection direction. And pausing and restarting a selection as described above works as expected.
+
+
+=== In some versions of Safari, the `editor.selection.isForward()` API could throw an exception due to an invalid selection
+
+When running within some versions of Safari, a bug in the `editor.selection.isForward()` API produced an invalid selection resulting in the `input` value provided to `input.parentNode.removeChild(input)` being `null`.
```

Comment:
This is incorrect and the second part of this sentence was about a test so shouldn't be in release notes.

So to be clear, the problem is that Safari itself can end up with an invalid selection and because of that it caused an exception to be thrown when calling `editor.selection.isForward()`. So TinyMCE now handles that invalid selection more gracefully, it didn't cause the invalid selection though.

Reply:
*`editor.selection.isForward()` API exception* section re-written, properly sheeting the blame home to Safari and not the API.